text/0000-pack-buildpack-new-alternatives.md

```diff
@@ -0,0 +1,206 @@
+
+# Motivation
+[motivation]: #motivation
+
+The creation of buildpacks in languages other than `bash` is undocumented in the [Buildpack Author Guide](https://buildpacks.io/docs/buildpack-author-guide).  In particular, creating a buildpack in Go using `libcnb` is undocumented because there is no mechanism to generate a scaffolded project.  This proposal adds a minimal Go/`libcnb` project template to `pack` and allows 3d parties to provide templates to create buildpacks for their chosen technology stack.
+
+`pack buildpack create` supports end-users who wish to scaffold a new bash or Go/`libcnb`-based buildpack.  The operation of `pack buildpack create` requires Internet access to succeed.  `pack buildpack create --template URL` allows projects (such as Paketo, Heroku and others) to define skeleton projects for new buildpacks.
+
+Replacing `pack buildpack new` with `pack buildpack create` allows buildpack authors to more-easily create new buildpacks in their chosen language and framework.
+
+# What it is
+[what-it-is]: #what-it-is
+
+Project scaffolding is [very](https://cookiecutter.readthedocs.io/) [popular](https://yeoman.io/) [in](https://github.com/kowainik/summoner) [other](https://crates.io/crates/cargo-scaffold) [ecosystems](https://github.com/golang-standards/project-layout).  Scaffolding systems are employed to ease onboarding of new developers.  Within `pack` this feature is targeted at onboarding buildpack authors.
+
+`pack buildpack create` should prompt the end user to choose an implementation language.  Thereafter, generate skeleton code according to the best practices for that language.  For example, users choosing Go as an implementation language would generate a [golang standard project layout](https://github.com/golang-standards/project-layout) and include `libcnb` best practices.  The layout would look similar to the following:
+
+```bash
+.
+├── buildpack.toml
+├── cmd
+│   └── main.go
+├── go.mod
+├── go.sum
+└── pkg
+    ├── build.go
+    └── detect.go
+```
+
+This approach is not opinionated on topics such as which testing framework to use.  Only components under github.com/buildpacks/* are used in the generated project.  `pack buildpack create --template URL` allows for more opinionated project scaffolding.  As such, we support new buildpack authors with onboarding and support experienced buildpack authors to use the scaffolding of their choice.
+
+# How it Works
+[how-it-works]: #how-it-works
+
+The design is modeled on [cookiecutter](https://cookiecutter.readthedocs.io/en/1.7.2/) with reference to [springerle](https://github.com/carlmjohnson/springerle) -- a similar implementation in golang -- and [create-go-app](https://github.com/create-go-app/).  We do we want to `os.Exec` a python subprocess to run cookiecutter as this would require availability of a python runtime environment.  Instead we propose to borrow heavily from create-go-app, and generate the default shell and libcnb skeletons from a cloned git repoistory.
```

Comment:
```suggestion
The design is modeled on [cookiecutter](https://cookiecutter.readthedocs.io/en/1.7.2/) with reference to [springerle](https://github.com/carlmjohnson/springerle) -- a similar implementation in golang -- and [create-go-app](https://github.com/create-go-app/).  We do not want to `os.Exec` a python subprocess to run cookiecutter as this would require availability of a python runtime environment.  Instead we propose to borrow heavily from create-go-app, and generate the default shell and libcnb skeletons from a cloned git repoistory.
```